camel/toolkits/search_toolkit.py

```
@@ -835,6 +835,102 @@ def search_baidu(self, query: str, max_results: int = 5) -> Dict[str, Any]:
+            }
+            html = requests.get(url, headers=headers)
+            html.encoding = 'utf-8'
+            soup = BeautifulSoup(html.text, 'html.parser')
+
+            b_results_element = soup.find("ol", id="b_results")
+            if b_results_element is None:
+                return {"results": []}
+
+            # Ensure b_results is a Tag and find all li elements
+            b_results_tag = cast(Tag, b_results_element)
+            result_items = b_results_tag.find_all("li")
+
+            results: List[Dict[str, Any]] = []
+            for i in range(min(len(result_items), max_results)):
+                row = result_items[i]
+                if not isinstance(row, Tag):
+                    continue
+
+                h2_element = row.find("h2")
+                if h2_element is None:
+                    continue
+                h2_tag = cast(Tag, h2_element)
+
+                title = h2_tag.get_text().strip()
+
+                link_tag_element = h2_tag.find("a")
+                if link_tag_element is None:
+                    continue
+                link_tag = cast(Tag, link_tag_element)
+
+                link = link_tag.get("href")
+                if link is None:
+                    continue
+
+                content_element = row.find("p", class_="b_algoSlug")
+                content_text = ""
+                if content_element is not None and isinstance(
+                    content_element, Tag
+                ):
+                    content_text = content_element.get_text()
+
+                row_data = {
+                    "result_id": i + 1,
+                    "snippet": content_text,
+                    "title": title,
+                    "link": link,
+                }
+                results.append(row_data)
+
+            if not results:
+                print(
+                    "Warning: No results found. Check "
+                    "if Bing HTML structure has changed."
+                )
```

use logger instead or print would be better, also we can provide this information to agent, so we can return this error message